tools/ansible/tasks/wsu/main.yaml

```diff
@@ -20,25 +20,151 @@
         msg: "Cluster not patched for hybrid overlay"
       when: "hybrid_query.stdout == ''"
 
+    - name: Get kubernetes version
+      shell: "oc version -o json | jq -r '.serverVersion.gitVersion'"
+      register: kubernetes_version
+
+    # kubernetes_version v1.17.2 should correspond to 1.17 for getting changelog url
+    - name: Get kubernetes version without patch version
+      shell: "version={{ kubernetes_version.stdout }} ; echo ${version%.*} | tr -d v"
+      register: kubernetes_version_without_patch
+
+    # Changelog url to lookup SHA for kube node binaries
+    # Example: https://github.com/kubernetes/kubernetes/blob/release-1.17/CHANGELOG/CHANGELOG-1.17.md
+    - name: Get kubernetes changelog url
+      shell: |
+        echo "https://github.com/kubernetes/kubernetes/blob/release-"\
+        "{{ kubernetes_version_without_patch.stdout }}/CHANGELOG/CHANGELOG-"\
+        "{{ kubernetes_version_without_patch.stdout }}.md"
+      register: changelog
+
+    - name: Get kube node binary SHA
+      shell: |
+        curl -s "{{ changelog.stdout }}" | \
+        grep -A 1 "{{ kubernetes_version.stdout }}" | \
+        grep -A 1 node-windows | tail -n 1 | sed -e 's/.*<code>\(.*\)<\/code>.*/\1/'
+      register: kube_node_sha
+
+    - name: Set kubelet location
+      set_fact:
+        # Example of kubelet download url: "https://dl.k8s.io/v1.16.2/kubernetes-node-windows-amd64.tar.gz"
+        kubelet_location: "https://dl.k8s.io/{{ kubernetes_version.stdout }}/kubernetes-node-windows-amd64.tar.gz"
+
+    - name: Get kubernetes minor version
+      shell: "oc version -o json | jq -r '.serverVersion.minor' | tr -d +"
```

Comment:
We are calling oc version multiple times, we should try and avoid that if possible, and parse the output of the original command as we need it.